src/main/java/zd/zero/waifu/motivator/plugin/assets/LocalStorageService.kt

```diff
@@ -0,0 +1,28 @@
+package zd.zero.waifu.motivator.plugin.assets
+
+import com.intellij.openapi.application.PathManager
+import com.intellij.openapi.diagnostic.Logger
+import java.io.IOException
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.*
+
+object LocalStorageService {
+    private val log = Logger.getInstance(this::class.java)
+
+    fun createDirectories(directoriesToCreate: Path) {
+        try {
+            Files.createDirectories(directoriesToCreate.parent)
+        } catch (e: IOException) {
+            log.error("Unable to create directories $directoriesToCreate for raisins", e)
+        }
+    }
+
+    fun getLocalAssetDirectory(): Optional<String> =
+        Optional.ofNullable(
+                PathManager.getConfigPath()
```

Comment:
The assets will be stored in a place that I know can be written to.
It's the configuration directory the IDE uses

![image](https://user-images.githubusercontent.com/15972415/88488631-1d0fd200-cf54-11ea-9b3f-e8fd79f6ad27.png)

Reply:
This is awesome, as long as the file is the same it won't be downloaded anymore, only the checksum. That's cool 🔥